File: api-code-examples/nodejs/docs-get.js
import assert from 'assert'
import { Iroh } from '@number0/iroh'

const node = await Iroh.memory({ enableDocs: true })
const nodeId = await node.net.nodeId()
console.log(`Started iroh node: ${nodeId}`)

const doc = await node.docs.create()
console.log(`Created doc: ${doc.id()}`)

// use the default author to write
const author = await node.authors.default();

// set a value in the document
const key = Array.from(Buffer.from("key"))
const value = Array.from(Buffer.from("value!"))
await doc.setBytes(author, key, value)

// get a key
const entry = await doc.getExact(author, key, false)

// read the value back
const res = await node.blobs.readToBytes(entry.hash)

// returned value matches the original
assert.deepEqual(value, res)

console.log(`Got value: ${Buffer.from(res).toString()}`)

File: api-code-examples/python/docs-get.py
import iroh
import asyncio


async def main():
    # Set options to enable docs
    options = iroh.NodeOptions()
    options.enable_docs = True

    # Create in memory iroh node
    node = await iroh.Iroh.memory_with_options(options)

    # Get default author
    author = await node.authors().default()

    # Create a document
    doc = await node.docs().create()
    print(f"Created document {doc.id()}")

    # Set content in the document
    key = b"python"
    hash = await doc.set_bytes(author, key, b"says hello")
    print(f"Inserted {hash}")

    # Get an entry from the document
    entry = await doc.get_exact(author, key, False)

    # Read content from the entry
    content = await entry.content_bytes(doc)
    print(f"Got content \"{content.decode('utf-8')}\"")

# Output:
# Created author huarctxgpvq2ucnifubjxvmac7c26evzudnynp5xrugkkm37ma7q
# Created document zmwwfsnnoxgij4q5bknfij5tpwbm2askypip3al3bahinucx65oq
# Inserted bafkr4ihasgdyqs6onufsjrmk5h5vcg2ud75u2iaokavwiulyg7wfno6fte
# Got content "says hello"

asyncio.run(main())
